scaleway/data_source_k8s_version.go

```diff
@@ -58,19 +58,35 @@ func dataSourceScalewayK8SVersionRead(ctx context.Context, d *schema.ResourceDat
 	if !ok {
 		return diag.FromErr(fmt.Errorf("could not find version %q", name))
 	}
-	res, err := k8sAPI.GetVersion(&k8s.GetVersionRequest{
-		Region:      region,
-		VersionName: name.(string),
-	}, scw.WithContext(ctx))
-	if err != nil {
-		return diag.FromErr(err)
+
+	var version *k8s.Version
+
+	if name == "latest" {
+		res, err := k8sAPI.ListVersions(&k8s.ListVersionsRequest{
+			Region: region,
+		}, scw.WithContext(ctx))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if len(res.Versions) > 1 {
+			version = res.Versions[0]
+		}
```

Comment:
```suggestion
		if len(res.Versions) == 0 {
			return diag.FromErr(fmt.Errorf("could not find the latest version"))
		}

		version = res.Versions[0]
```

Should never happen but we never know